core/src/main/java/com/linecorp/armeria/server/cors/CorsConfig.java

```diff
@@ -106,7 +106,7 @@ public CorsPolicy getPolicy(@Nullable String origin, RoutingContext routingConte
                 isPathMatched(policy, routingContext)) {
                 return policy;
             } else if (!isNullOrigin && isPathMatched(policy, routingContext)) {
-                if (policy.originPredicate().test(origin)) {
+                if (policy.originPredicate().test(lowerCaseOrigin)) {
```

Comment:
Makes sense since domain names are [case-insensitive](https://www.rfc-editor.org/rfc/rfc4343.txt#:~:text=Domain%20Name%20System%20(DNS)%20names%20are%20%22case%20insensitive%22).
Should we cast `origin` to lowercase in the WebSocketService implementation?
https://github.com/line/armeria/blob/d31dc60bb080a51915e528473c6d8c96d5c699eb/core/src/main/java/com/linecorp/armeria/internal/server/websocket/DefaultWebSocketService.java#L286-L286

Reply:
Additionally, we also need to cast `allowedOrigins` in lowercase.
https://github.com/line/armeria/blob/d31dc60bb080a51915e528473c6d8c96d5c699eb/core/src/main/java/com/linecorp/armeria/server/websocket/WebSocketServiceBuilder.java#L187

Reply:
I actually interpret the specification to be case-insensitive.
ref: https://www.w3.org/TR/2020/SPSD-cors-20200602/#resource-requests
> If the value of the [Origin](https://www.w3.org/TR/2020/SPSD-cors-20200602/#http-origin) header is not a [case-sensitive](https://www.w3.org/TR/2020/SPSD-cors-20200602/#case-sensitive) match for any of the values in [list of origins](https://www.w3.org/TR/2020/SPSD-cors-20200602/#list-of-origins), do not set any additional headers and terminate this set of steps.

Having said this, I agree consistency is important. Updated

Reply:
I didn't know that it is case-sensitive. 😅 Thanks for the link.

> I agree consistency is important.
+1 Case-insensive matching does not seem to be a security problem.